evap/staff/views.py

```diff
@@ -1386,23 +1388,67 @@ def evaluation_email(request, evaluation_id):
     )
 
 
-def helper_delete_users_from_evaluation(evaluation, operation):
-    if "participants" in operation:
-        deleted_person_count = evaluation.participants.count()
-        deletion_message = _("{} participants were deleted from evaluation {}")
-        evaluation.participants.clear()
+class ImportAction(Enum):
+    COPY = "copy"
+    IMPORT = "import"
+    TEST = "test"
+
+    @classmethod
+    def from_operation(cls, operation: str) -> "ImportAction":
+        if operation.startswith("copy-"):
+            return cls.COPY
+        if operation.startswith("import-"):
+            return cls.IMPORT
+        if operation.startswith("test-"):
+            return cls.TEST
+        raise ValueError(f"Unknown operation: {operation}")
+
+
+def import_or_copy_participants(
+    request,
+    replace: bool,
+    import_action: Literal[ImportAction.COPY, ImportAction.IMPORT],
+    import_type: ImportType,
+    evaluation: Evaluation,
+    copy_form: EvaluationParticipantCopyForm,
+) -> bool:
+    if replace:
+        if import_type == ImportType.PARTICIPANT:
+            deleted_person_count = evaluation.participants.count()
+            deletion_message = _("{} participants were deleted from evaluation {}")
+            evaluation.participants.clear()
+        elif import_type == ImportType.CONTRIBUTOR:
+            deleted_person_count = evaluation.contributions.exclude(contributor=None).count()
+            deletion_message = _("{} contributors were deleted from evaluation {}")
+            evaluation.contributions.exclude(contributor=None).delete()
+    if import_action == ImportAction.IMPORT:
+        file_content = get_import_file_content_or_raise(request.user.id, import_type)
+        importer_log = import_persons_from_file(import_type, evaluation, test_run=False, file_content=file_content)
+        delete_import_file(request.user.id, import_type)
+    elif import_action == ImportAction.COPY:
+        copy_form.evaluation_selection_required = True
+        if not copy_form.is_valid():
+            return False
+        import_evaluation = copy_form.cleaned_data["evaluation"]
+        importer_log = import_persons_from_evaluation(
+            import_type, evaluation, test_run=False, source_evaluation=import_evaluation
+        )
     else:
-        assert "contributors" in operation
-        deleted_person_count = evaluation.contributions.exclude(contributor=None).count()
-        deletion_message = _("{} contributors were deleted from evaluation {}")
-        evaluation.contributions.exclude(contributor=None).delete()
+        assert_never(import_action)
 
-    return deleted_person_count, deletion_message
+    if replace:
+        importer_log.add_success(  # pylint: disable=possibly-used-before-assignment  # see https:www.//github.com/pylint-dev/pylint/issues/9780
```

Comment:
I don't understand the exclusion. If `replace == True` but `import_type` is any other value than `PARTICIPANT` or `CONTRIBUTOR`, the variable `deletion_message` will not be initialized here. I guess it's just incidental that we expect no other import type? But that's nothing pylint can see, we should just `assert` that then?

(The URL is invalid, misplaced `//`)

Reply:
but the ci said: `evap/staff/views.py:1440:8: E0606: Possibly using variable 'importer_log' before assignment (possibly-used-before-assignment)`. However, the case you are describing seems to be a separate problem

Reply:
Ok, I agree that pylint triggering for `importer_log` here is a false positive. I was just mentally checking the full statement :D

Reply:
```suggestion
        importer_log.add_success(  # pylint: disable=possibly-used-before-assignment  # see https://www.github.com/pylint-dev/pylint/issues/9780
```